ctera/src/main/java/ch/cyberduck/core/ctera/CteraAclPermissionFeature.java

```diff
@@ -0,0 +1,198 @@
+     * This permission does not automatically allow running program files.
+     * Directories only.
+     */
+    public static final Acl.Role TRAVERSEPERMISSION = new Acl.Role("traversepermission");
+
+    /**
+     * Create Files: Allows or denies creating files within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEFILEPERMISSION = new Acl.Role("Createfilepermission");
+
+    /**
+     * Create Folders: Allows or denies creating subfolders within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEDIRECTORIESPERMISSION = new Acl.Role("CreateDirectoriespermission");
+
+    static final List<Acl.Role> allCteraCustomACLRoles = Collections.unmodifiableList(Arrays.asList(
+            READPERMISSION, WRITEPERMISSION, EXECUTEPERMISSION, DELETEPERMISSION, TRAVERSEPERMISSION, CREATEFILEPERMISSION, CREATEDIRECTORIESPERMISSION
+    ));
+
+    final static List<QName> allCteraCustomACLQn = Collections.unmodifiableList(allCteraCustomACLRoles.stream().map(CteraAclPermissionFeature::toQn).collect(Collectors.toList()));
+
+    public static QName toQn(final Acl.Role role) {
+        return new QName(CTERA_NAMESPACE_URI, role.getName(), CTERA_NAMESPACE_PREFIX);
+    }
+
+    public static String toProp(final QName qn) {
+        return qn.getLocalPart();
+    }
+
+    public static Acl.Role toRole(final QName qn) {
+        return new Acl.Role(toProp(qn));
+    }
+
+    public static Acl customPropsToAcl(final Map<String, String> customProps) {
+        // TODO CTERA-136 distinguish empty and null custom props?
+        if(customProps.isEmpty()) {
+            // ignore acl in preflight
+            return Acl.EMPTY;
+        }
+        final Acl acl = new Acl();
+        // initialize with no roles
+        acl.addAll(new Acl.CanonicalUser());
+        for(final QName qn : allCteraCustomACLQn) {
+            if(customProps.containsKey(toProp(qn))) {
+                final String val = customProps.get(toProp(qn));
+                if(Boolean.parseBoolean(val)) {
+                    acl.addAll(new Acl.CanonicalUser(), toRole(qn));
+                }
+            }
+        }
+        return acl;
+    }
+
+    private final DAVSession session;
+
+    public CteraAclPermissionFeature(final DAVSession session) {
+        this.session = session;
+    }
+
+    @Override
+    public Acl getPermission(final Path file) throws BackgroundException {
+        final PathAttributes attributes = new CteraAttributesFinderFeature(session).find(file, new DisabledListProgressListener());
+        return attributes.getAcl();
+    }
+
+    @Override
+    public void setPermission(final Path file, final Acl acl) throws BackgroundException {
+        try {
+            List<Element> setProps = this.toCustomProperties(acl);
+            session.getClient().patch(new DAVPathEncoder().encode(file), setProps, Collections.emptyList(), Collections.emptyMap());
```

Comment:
Add test for setting ACLs. I don't expect this to be supported.

Reply:
47fcf4fa64bb98aaa3e9819e8eb2197354b6b1bd